jerry-core/api/jerry-snapshot.h

```diff
@@ -24,18 +24,39 @@
 typedef struct
 {
   /* The size of this structure is recommended to be divisible by
-   * JMEM_ALIGNMENT. Otherwise some bytes after the header are wasted. */
+   * uint32_t alignment. Otherwise some bytes after the header are wasted. */
+  uint32_t magic; /**< four byte magic number */
   uint32_t version; /**< version number */
-  uint32_t lit_table_offset; /**< offset of the literal table */
-  uint32_t lit_table_size; /**< size of literal table */
-  uint32_t is_run_global; /**< flag, indicating whether the snapshot
-                            *   was saved as 'Global scope'-mode code (true)
-                            *   or as eval-mode code (false) */
+  uint32_t configuration; /**< configuration flags */
+  uint32_t lit_table_offset; /**< byte offset of the literal table */
+  uint32_t number_of_funcs; /**< number of primary ECMAScript functions */
+  uint32_t func_offsets[1]; /**< function offsets (lowest bit: global(0) or eval(1) context) */
 } jerry_snapshot_header_t;
 
 /**
- * Jerry snapshot format version
+ * Evaluate this function on the top of the scope chain.
  */
-#define JERRY_SNAPSHOT_VERSION (7u)
+#define JERRY_SNAPSHOT_EVAL_CONTEXT 0x1u
+
+/**
+ * Jerry snapshot magic marker.
+ */
+#define JERRY_SNAPSHOT_MAGIC (0x5453534Au)
```

Comment:
This is "JSST", if I'm right. Not the most important thing, but I'd open a discussion on this, what the magic should be. To me, the prefix "JS" is too generic, not JerryScript-specific. So, I'd vote for something like "JRSS" -- but that's just a minor comment.

Reply:
The magic string also encodes the alignment, so the the center characters should be the same to allow finding this easier.

Reply:
I don't follow this reasoning. These are four bytes. "JRSS" is just as easy to find and differentiate from "SSRJ" than "JSST" vs "TSSJ". As long as the magic word is not a palindrome, anything can be used. (Assuming that you've meant byte-order or endianness instead of alignment.)

Other variants: JRST, JRRY

Reply:
JRRY is ok, but I would still prefer JSST.

If I understood correctly, the request was recognizing jerry snapshot in random binary data with naked eye (and not explicitly searching for it). This is easier if the two center letters are the same, and the first and last looks similar. J and T shares a vertical pipe (I). J and Y are not that bad, but perhaps OSS0 is the best.

Reply:
I'd like to hear the opinion of others. @martijnthe you were the first to mention magic bytes. What's your vote?

Reply:
You're taking this very seriously :D

I think most people are working with little endian architectures, I would "optimize" for that.
I don't think people will be looking at a mix of little endian and big endian dumps, so I think making the first and last letter the same would probably not matter much, really.

I'd vote for `JRRY` (as displayed in a memory dump of a little-endian machine) just because it looks most like Jerry.
So I think that means the value in the source would have to be `0x4A525259`.